stream doesn't send close event

gsick opened the pull request:
Hi,

In process:exit() the callback onClose is never call because stream doesn't send "close" event but "end" event.
I didn't change the version...

Thanks

rphillips commented:
in this case, the stdout/stderr streams should emit a close event because the underlying socket is a tty

```lua
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process:exit(0)
```

rphillips commented:
Make sure you are using 2.2.2 as well.

gsick commented:
I'm using the master branch. I cleaned everything and checkout master.

I added a simple trace in the callback onClose:
```lua
local function exit(self, code)
  ...
  local function onClose()
    print("onClose")
   ...
    self:emit('exit', code)
    os.exit(code)
  end
  ...
end
```

After I execute your example:
```lua
process.stdout:write('hello world\n')
process.stdout:write('hello world\n')
process:exit(0)
```

the output is:
```bash
hello world
hello world
```

the onClose cb is not executed

rphillips commented:
@gsick could you try this branch out for me `fixes/fix_stdout_stderr_wait_for_finish` and let me know?

gsick commented:
@rphillips Tested and it works like a charm.
